frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -35,6 +40,77 @@ export interface ClusterRequest {
   certificateAuthorityData?: string;
 }
 
+//refreshToken checks if the token is about to expire and refreshes it if so.
+async function refreshToken(token: string | null) {
+  if (!token || TOKEN_REFRESH_INPROGRESS) {
+    return;
+  }
+  // decode token
+  const decodedToken: any = decodeToken(token);
+
+  // return if the token doesn't have an expiry time
+  if (!decodedToken.exp) {
+    return;
+  }
+  // convert expiry seconds to date object
+  const expiry = decodedToken.exp;
+  const now = new Date().valueOf();
+  const expDate = new Date(0);
+  expDate.setUTCSeconds(expiry);
+
+  // calculate time to expiry in minutes
+  const diff = (expDate.valueOf() - now) / 60000;
+  // If the token is not about to expire return
+  if (diff > MIN_LIFESPAN_FOR_TOKEN_REFRESH) {
+    return;
+  }
+  let namespace = '';
+  let serviceAccountName = '';
+  if (decodedToken && decodedToken['kubernetes.io']) {
+    const tokenKubeInfo = decodedToken['kubernetes.io'];
+    namespace = tokenKubeInfo['namespace'] || '';
+    const serviceAccount = tokenKubeInfo['serviceaccount'] || {};
+    serviceAccountName = serviceAccount['name'] || '';
+  }
+  const cluster = getCluster();
+  if (!cluster || namespace === '' || serviceAccountName === '') {
+    return;
+  }
+
+  console.debug('Refreshing token');
+  TOKEN_REFRESH_INPROGRESS = true;
+
+  let tokenUrl = combinePath(BASE_HTTP_URL, `/${CLUSTERS_PREFIX}/${cluster}`);
+  tokenUrl = combinePath(
+    tokenUrl,
+    `api/v1/namespaces/${namespace}/serviceaccounts/${serviceAccountName}/token`
+  );
+  const tokenData = {
+    kind: 'TokenRequest',
+    apiVersion: 'authentication.k8s.io/v1',
+    metadata: { creationTimestamp: null },
+    spec: { expirationSeconds: 86400 },
+  };
+
+  const response = await fetch(tokenUrl, {
```

Comment:
There should be some error handling with fetch: See https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful

Needs a try/catch or a .catch() added.

Reply:
added